rust/src/message/message.rs

```diff
@@ -0,0 +1,178 @@
+use anyhow::{anyhow, Context};
+use std::borrow::Borrow;
+
+use crate::{
+    certificate::Certificate,
+    crypto::{ByteObject, PublicEncryptKey, SecretEncryptKey, SecretSigningKey, Signature},
+    mask::{Mask, MaskedModel},
+    message::{
+        DecodeError,
+        FromBytes,
+        Header,
+        HeaderOwned,
+        Payload,
+        PayloadOwned,
+        Sum2Owned,
+        SumOwned,
+        Tag,
+        ToBytes,
+        UpdateOwned,
+    },
+    LocalSeedDict,
+};
+
+/// A message
+pub struct Message<C, D, M, N> {
+    /// Message header
+    pub header: Header<C>,
+    /// Message payload
+    pub payload: Payload<D, M, N>,
+}
+
+pub type MessageOwned = Message<Certificate, LocalSeedDict, MaskedModel, Mask>;
```

Comment:
Exactly. Although in the end removed the `Borrowed` version, because references won't cut it if  we want to do the serialization in a separate transport layer. In such cases we'll probably have to use a custom `Message` that uses `Arc`s, internally.